internal/auth/oidc/repository_auth_method_update.go

```diff
@@ -17,32 +31,561 @@ import "context"
+	}
+
+	if err := am.isComplete(); err != nil {
+		return errors.Wrap(err, op)
+	}
+
+	// FYI: once converted to an oidc.Provider, any certs configured will be used as trust anchors for all HTTP requests
+	provider, err := convertToProvider(ctx, am)
+	if err != nil {
+		return errors.Wrap(err, op)
+	}
+
+	info, err := provider.DiscoveryInfo(ctx)
+	if err != nil {
+		return errors.Wrap(err, op)
+	}
+
+	var result *multierror.Error
+	if info.Issuer != am.DiscoveryUrl {
+		result = multierror.Append(result, errors.New(errors.InvalidParameter, op,
+			fmt.Sprintf("auth method issuer doesn't match discovery issuer: expected %s and got %s", am.DiscoveryUrl, info.Issuer)))
+	}
+	for _, a := range am.SigningAlgs {
+		if !strutil.StrListContains(info.IdTokenSigningAlgsSupported, a) {
+			result = multierror.Append(result, errors.New(errors.InvalidParameter, op,
+				fmt.Sprintf("auth method signing alg is not in discovered supported algs: expected %s and got %s", a, info.IdTokenSigningAlgsSupported)))
+		}
+	}
+	providerClient, err := provider.HTTPClient()
+	if err != nil {
+		result = multierror.Append(result, errors.New(errors.Unknown, op, "unable to get oidc http client", errors.WithWrap(err)))
+		return result.ErrorOrNil()
+	}
+
+	// we need to prevent redirects during these tests... we don't want to have
+	// redirects going to the controller's callback (aka the configured provider's callback)
+	providerClient.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+
+	// test JWKs URL
+	statusCode, err := pingEndpoint(ctx, providerClient, "JWKs", "GET", info.JWKSURL)
+	if err != nil {
+		result = multierror.Append(result, errors.New(errors.Unknown, op, fmt.Sprintf("unable to verify JWKs endpoint: %s", info.JWKSURL), errors.WithWrap(err)))
+	}
+	if statusCode != 200 {
+		result = multierror.Append(result, errors.New(errors.Unknown, op, fmt.Sprintf("non-200 status (%d) from JWKs endpoint: %s", statusCode, info.JWKSURL), errors.WithWrap(err)))
+	}
+
+	// test Auth URL
+	if _, err := pingEndpoint(ctx, providerClient, "AuthURL", "GET", info.AuthURL); err != nil {
+		result = multierror.Append(result, errors.New(errors.Unknown, op, fmt.Sprintf("unable to verify authorize endpoint: %s", info.AuthURL), errors.WithWrap(err)))
+	}
+
+	// test Token URL
+	if _, err := pingEndpoint(ctx, providerClient, "TokenURL", "POST", info.TokenURL); err != nil {
+		result = multierror.Append(result, errors.New(errors.Unknown, op, fmt.Sprintf("unable to verify token endpoint: %s", info.TokenURL), errors.WithWrap(err)))
+	}
+
+	// we're not verifying the UserInfo URL, since it's not a required dependency.
+
+	return result.ErrorOrNil()
+}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// pingEndpoint will make an attempted http request, return status code and errors
+// (non 2xx status codes are not an error)
```

Comment:
FWIW, my immediate read on this comment was "wouldn't 4xx be an error" but because actually having a comment about HTTP semantics made me think that the error for this function had to do with HTTP semantics. I'd just remove this line -- saying it returns the status code and method error seems fine. :-)

Reply:
done.